### DISCUSSION TITLE
Forgot password, recovery email is not arriving

### DISCUSSION THREAD

**Discussion by Ascar699** [chatwoot/chatwoot#4939] [Q&A] [answered]
Hello,
I installed Chatwoot a while ago in a virtual machine and left it for long enough to forget the password I used to register admin user. 

Now, I want to log in to learn more how to use Chatwoor but I don't receive any email after clicking he Forgot Password button. When I tried using some other email I received an error saying no such user. Is there a way to check the password from inside the virtual machine / docker image?

Thanks

**Comment by vishnu-narayanan** [comment, marked as answer]
@Mantukai Do you remember setting up `SMTP` variables? It is needed for forgot password emails to work. You could open a shell in the chatwoot docker image, access the ruby console and reset the password.

```
docker exec -it <chatwoot-rails-continer-name> -c 'RAILS_ENV=production bundle exec rails c'
3.0.4 :001 > User.count
 => 1
3.0.4 :002 > user=User.find_by(email: "email@email.com")
3.0.4 :003 > user.password="NewPassword1!"
 => "NewPassword1!"
3.0.4 :004 > user.save!
```
ref: https://www.chatwoot.com/docs/self-hosted/deployment/docker#running-rails-console

**Reply by Zion7899** [reply]
Are you on snap?

**Comment by Ascar699** [comment]
@vishnu-narayanan Thank you for your suggestion.  I abandoned my Chatwoot instance shortly after I installed and 99% I did not do any SMTP setup.

It looks like Chatwoot was not installed as a Docker image.  At least when I ssh into the vm and run docker images I get "Command 'docker' not found, but can be installed with:"  Is it possible that Chatwoot gets installed as a deb package or just directly to the file system?  I can at least tell that I have chatwoot user directory under /home.  If yes, please tell  me where and I how I can retrieve user password or reset it.

Thanks very much in advance

**Comment by Ascar699** [comment]
@vishnu-narayanan I followed your instructions inside the chatwoot directory and succeeded in resetting the password.  Thank you.
I figured that I installed directly on Ubuntu without Docker.
